json-schema-core/src/main/java/io/ballerina/jsonschema/core/Generator.java

```diff
@@ -167,11 +230,504 @@ String convert(Object schemaObject, String name) throws Exception {
                 })
                 .collect(Collectors.joining(PIPE));
 
-        ID_TO_TYPE_MAP.put(schema.getIdKeyword(), typeName);
         return typeName;
     }
 
-    private String generateStringRepresentation(Object obj) throws InvalidDataTypeException {
+    private String createType(String name, Schema schema, Object type) throws Exception {
+        if (type == null) {
+            return NULL;
+        }
+        if (type == Boolean.class) {
+            return BOOLEAN;
+        }
+        if (type == Long.class) {
+            return createInteger(name, schema.getMinimum(), schema.getExclusiveMinimum(), schema.getMaximum(),
+                    schema.getExclusiveMaximum(), schema.getMultipleOf());
+        }
+        if (type == Double.class) {
+            return createNumber(name, schema.getMinimum(), schema.getExclusiveMinimum(), schema.getMaximum(),
+                    schema.getExclusiveMaximum(), schema.getMultipleOf());
+        }
+        if (type == String.class) {
+            return createString(name, schema.getFormat(), schema.getMinLength(), schema.getMaxLength(),
+                    schema.getPattern());
+        }
+        if (type == ArrayList.class) {
+            return createArray(name, schema.getPrefixItems(), schema.getItems(), schema.getContains(),
+                    schema.getMinItems(), schema.getMaxItems(), schema.getUniqueItems(), schema.getMaxContains(),
+                    schema.getMinContains(), schema.getUnevaluatedItems());
+        }
+        if (type == Map.class) {
+            return createObject(name, schema.getAdditionalProperties(), schema.getProperties(),
+                    schema.getPatternProperties(), schema.getDependentSchema(), schema.getPropertyNames(),
+                    schema.getUnevaluatedProperties(), schema.getMaxProperties(), schema.getMinProperties(),
+                    schema.getDependentRequired(), schema.getRequired());
+        }
+        throw new RuntimeException("Type currently not supported");
+    }
+
+    private String createInteger(String name, Double minimum, Double exclusiveMinimum, Double maximum,
+                                       Double exclusiveMaximum, Double multipleOf) {
+        if (isCustomTypeNotRequired(minimum, exclusiveMinimum, maximum, exclusiveMaximum, multipleOf)) {
+            return INTEGER;
+        }
+
+        if (isNumberLimitInvalid(minimum, exclusiveMinimum, maximum, exclusiveMaximum)) {
+            return NEVER;
+        }
+
+        addImports(BAL_JSON_DATA_MODULE, this);
```

Comment:
```suggestion
        addImport(BAL_JSON_DATA_MODULE, this);
```

Can also be overloaded/two methods to avoid passing `this` in the common case.

Reply:
Can also have an addJsonDataImport method.